ffi/derive/src/impl_visitor.rs

```diff
@@ -188,9 +199,13 @@ impl<'ast> ImplVisitor<'ast> {
 }
 
 impl<'ast> FnVisitor<'ast> {
-    pub const fn new(self_ty: Option<&'ast syn::Path>) -> Self {
+    pub const fn new(
+        self_ty: Option<&'ast syn::Path>,
```

Comment:
Prefer passing the value directly, unwrap options before use.

Reply:
I tried your suggestion and i think resulting code is less convenient.
In case of freestanding functions `self_ty` is `None` and this is intended.